prov/verbs/src/verbs_cm.c

```diff
@@ -190,11 +190,15 @@ vrb_msg_ep_connect(struct fid_ep *ep_fid, const void *addr,
 
 	ofi_genlock_lock(&vrb_ep2_progress(ep)->ep_lock);
 	assert(ep->state == VRB_IDLE);
-	ep->state = VRB_RESOLVE_ROUTE;
-	ret = rdma_resolve_route(ep->id, VERBS_RESOLVE_TIMEOUT);
-	if (ret) {
+	ep->state = VRB_RESOLVE_ADDR;
+	if (rdma_resolve_addr(ep->id, ep->info_attr.src_addr,
+			      ep->info_attr.dest_addr, VERBS_RESOLVE_TIMEOUT)) {
```

Comment:
What's the pros and cons calling `rdma_resovle_addr` here vs when the endpoint is enabled? The EQ should have already been bound to the endpoint when the endpoint is enabled.

Reply:
I believe that calling `rdma_resolve_addr()` when the endpoint is enabled would add more complexity to the code: `fi_connect()` would have to wait for the event `RDMA_CM_EVENT_ADDR_RESOLVED` before it can resume the connection establishment process (rdma_resolve_route() + rdma_connect()).
Having all the CM logic inside the EQ makes the code cleaner IMHO.

Reply:
Until rdma_resolve_addr() completes, isn't necessarily bound to a verbs device.  That's likely the con -- we don't have a device, don't have a qp, can't post receives...